src/bundle.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from "path";
+import { SourceNode } from "source-map";
+import * as ts from "typescript";
+import { couldNotFindBundleEntryPoint } from "./diagnostics";
+import { resolveFromRootDir } from "./resolve";
+import { EmitHost, TranspiledFile } from "./Transpile";
+import { formatPathToLuaPath, trimExtension } from "./utils";
+
+const formatPath = (path: string) => formatPathToLuaPath(trimExtension(path));
+
+export function bundleTranspiledFiles(
+    bundleFile: string,
+    entryModule: string,
+    transpiledFiles: TranspiledFile[],
+    program: ts.Program,
+    emitHost: EmitHost
+): [ts.Diagnostic[], TranspiledFile] {
+    const diagnostics: ts.Diagnostic[] = [];
+
+    const resolvedEntryModule = resolveFromRootDir(program, entryModule);
+    if (!transpiledFiles.some(f => resolveFromRootDir(program, f.fileName) === resolvedEntryModule)) {
```

Comment:
Is `resolveFromRootDir` required? Isn't `fileName` always resolved and normalized?

Reply:
Don't think it is when you manually construct the program (like we do in our tests)